corehq/messaging/smsbackends/icds/models.py

```diff
@@ -0,0 +1,130 @@
+import sys
+from urllib import urlencode
+from urllib2 import urlopen
+
+from crispy_forms import layout as crispy
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.sms.mixin import BackendProcessingException
+from corehq.apps.sms.models import SQLSMSBackend
+from corehq.apps.sms.forms import BackendForm
+from dimagi.utils.django.fields import TrimmedCharField
+
+ERROR_CODES = {
+    "-2": "Invalid credentials",
```

Comment:
Curious about the number-in-a-string format here. Seems like the type of thing that could be easy to mess up (like trying `"-1" == -1` by mistake), but it looks like you've got a system.